parser/xml/src/main/java/org/codice/ddf/parser/xml/XmlParser.java

```diff
@@ -0,0 +1,180 @@
+                marshaller.setEventHandler(configurator.getHandler());
+            }
+            for (Map.Entry<String, Object> propRow : configurator.getProperties().entrySet()) {
+                marshaller.setProperty(propRow.getKey(), propRow.getValue());
+            }
+
+            marshaller.marshal(obj, os);
+        } catch (JAXBException e) {
+            LOGGER.error("Error marshalling ", e);
+            throw new ParserException("Error marshalling", e);
+        } finally {
+            Thread.currentThread().setContextClassLoader(tccl);
+        }
+    }
+
+    @Override
+    public <T> T unmarshal(ParserConfigurator configurator, Class<? extends T> cls,
+            InputStream stream) throws ParserException {
+        JAXBContext jaxbContext = getContext(configurator.getContextPath(),
+                configurator.getClassLoader());
+
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(configurator.getClassLoader());
+
+            XMLStreamReader xmlStreamReader = XMLInputFactory.newInstance()
+                    .createXMLStreamReader(stream);
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            if (configurator.getAdapter() != null) {
+                unmarshaller.setAdapter(configurator.getAdapter());
+            }
+            if (configurator.getHandler() != null) {
+                unmarshaller.setEventHandler(configurator.getHandler());
+            }
+            for (Map.Entry<String, Object> propRow : configurator.getProperties().entrySet()) {
+                unmarshaller.setProperty(propRow.getKey(), propRow.getValue());
+            }
+
+            return (T) unmarshaller.unmarshal(xmlStreamReader);
+        } catch (JAXBException | XMLStreamException e) {
+            LOGGER.error("Error unmarshalling ", e);
+            throw new ParserException("Error unmarshalling", e);
+        } finally {
+            Thread.currentThread().setContextClassLoader(tccl);
+        }
+    }
+
+    private JAXBContext getContext(List<String> contextPath, ClassLoader loader)
+            throws ParserException {
+        String joinedPath = CTX_JOINER.join(contextPath);
+
+        JAXBContext jaxbContext;
+
+        try {
+            jaxbContext = jaxbContextCache.get(new CacheKey(joinedPath, loader));
+        } catch (ExecutionException e) {
+            LOGGER.error("Unable to create JAXB context using context path: {}", joinedPath, e);
+            throw new ParserException("Unable to create XmlParser", e.getCause());
+        }
+
+        return jaxbContext;
+    }
+
+    static class CacheKey {
+        private final String joinedPath;
+
+        private final ClassLoader loader;
+
+        CacheKey(String joinedPath, ClassLoader loader) {
+            this.joinedPath = joinedPath;
+            this.loader = loader;
+        }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
Might consider using Apache Commons Lang to build the equals and hashCode.

https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/builder/EqualsBuilder.html
https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/builder/HashCodeBuilder.html